pkg/postage/postagecontract/contract.go

```diff
@@ -0,0 +1,187 @@
+package postagecontract
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/postage"
+	"github.com/ethersphere/bee/pkg/postage/listener"
+	"github.com/ethersphere/bee/pkg/settlement/swap/transaction"
+	"github.com/ethersphere/sw3-bindings/v2/simpleswapfactory"
+)
+
+var (
+	postageStampABI   = parseABI(listener.PostageStampABIJSON)
+	erc20ABI          = parseABI(simpleswapfactory.ERC20ABI)
+	batchCreatedTopic = postageStampABI.Events["BatchCreated"].ID
+
+	ErrBatchCreationFailed = errors.New("batch creation failed")
+)
+
+type Interface interface {
```

Comment:
single method interfaces in go are usually called after the name of the function + `er`.
`io.Reader`, `Closer`, `LoadSaver`, etc... so `BatchCreater`  (i know, its a typo)

Reply:
my idea of this package was more to put all of the postage stamp (write) code there, as the code for topup and alike will also look similar and require the same helper code. then the interface would not be a single method anymore.

Reply:
ok